src/overlays/actors/ovl_En_GeldB/z_en_geldb.c

```diff
@@ -33,93 +97,1468 @@ const ActorInit En_GeldB_InitVars = {
+                } else {
+                    EnGeldB_SetupDamaged(this);
+                }
+            }
+        }
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_GeldB/func_80A37F98.s")
+void EnGeldB_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
+    EnGeldB* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_GeldB/func_80A38054.s")
+    func_80A392D8(this, globalCtx);
+    if (this->actor.colChkInfo.damageEffect != GELDB_DAMAGE_UNK6) {
+        Actor_MoveForward(&this->actor);
+        func_8002E4B4(globalCtx, &this->actor, 15.0f, 30.0f, 60.0f, 0x1D);
+        this->actionFunc(this, globalCtx);
+        this->actor.posRot2.pos = this->actor.posRot.pos;
+        this->actor.posRot2.pos.y += 40.0f;
+        func_80A391D8(this, globalCtx);
+    }
+    Collider_CylinderUpdate(&this->actor, &this->bodyCollider);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->bodyCollider.base);
+    if ((this->actionState >= GELDB_READY) && (this->spinAttackState < 2) &&
+        ((this->actor.dmgEffectTimer == 0) || !(this->actor.dmgEffectParams & 0x4000))) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->bodyCollider.base);
+    }
+    if ((this->actionState == GELDB_BLOCK) && (this->skelAnime.animCurrentFrame == 0.0f)) {
+        CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->blockCollider.base);
+    }
+    if (this->swordState > 0) {
+        CollisionCheck_SetAT(globalCtx, &globalCtx->colChkCtx, &this->swordCollider.base);
+    }
+    if (this->blinkState == 0) {
+        if ((Math_Rand_ZeroOne() < 0.1f) && ((globalCtx->gameplayFrames % 4) == 0)) {
+            this->blinkState++;
+        }
+    } else {
+        this->blinkState = (this->blinkState + 1) & 3;
+    }
+}
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_GeldB/func_80A380EC.s")
+s32 EnGeldB_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot,
+                             void* thisx) {
+    EnGeldB* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/actors/ovl_En_GeldB/func_80A38290.s")
+    OPEN_DISPS(globalCtx->state.gfxCtx, "../z_en_geldB.c", 2507);
+    if (limbIndex == 3) {
+        rot->z += this->headRot.x;
+        rot->x += this->headRot.y;
+        rot->y += this->headRot.z;
+    } else if (limbIndex == 6) {
+        gDPPipeSync(POLY_OPA_DISP++);
+        gDPSetEnvColor(POLY_OPA_DISP++, 80, 60, 10, 255);
+    } else if ((limbIndex == 11) || (limbIndex == 16)) {
+        gDPPipeSync(POLY_OPA_DISP++);
+        gDPSetEnvColor(POLY_OPA_DISP++, 140, 170, 230, 255);
+        gDPSetPrimColor(POLY_OPA_DISP++, 0, 0, 255, 255, 255, 255);
+    } else {
+        gDPPipeSync(POLY_OPA_DISP++);
+        gDPSetEnvColor(POLY_OPA_DISP++, 140, 0, 0, 255);
+    }
+    CLOSE_DISPS(globalCtx->state.gfxCtx, "../z_en_geldB.c", 2529);
+    return false;
```

Comment:
```suggestion
    return 0;
```

Reply:
oh, missed this

Reply:
OverrideLimbDraw actually does return a boolean.

Reply:
we havent been treating it as one so far. even if this has changed with the new updates, it will be updated at the same time